applications/nrf5340_audio/src/modules/display/tab_menu.c

```diff
@@ -0,0 +1,353 @@
+	lv_obj_set_style_radius(button, LV_RADIUS_CIRCLE, 0);
+	lv_obj_align(button, button_position, 0, 0);
+
+	lv_obj_t *label = lv_label_create(button);
+	lv_label_set_text(label, label_text);
+	lv_obj_set_style_text_font(label, font_size, LV_STATE_DEFAULT);
+	lv_obj_align(label, LV_ALIGN_CENTER, 0, 0);
+	lv_obj_add_event_cb(button, button_callback_function, LV_EVENT_CLICKED, NULL);
+
+	lv_obj_add_style(button, &style_btn_red, LV_STATE_PRESSED);
+
+	return label;
+}
+
+static void play_pause_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_PLAY_PAUSE;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void volume_up_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_VOLUME_UP;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void volume_down_button_event_cb(lv_event_t *e)
+
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_VOLUME_DOWN;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void btn4_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_4;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void btn5_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_5;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void vu_bar_update(int vu_val)
+{
+	lv_bar_set_value(vu_bar, vu_val, LV_ANIM_ON);
+}
+
+static int streaming_state_update(lv_obj_t *play_pause_button)
+{
+	int ret;
+	const struct zbus_channel *chan;
+
+	ret = zbus_sub_wait(&le_audio_evt_sub_display, &chan, K_NO_WAIT);
+
+	if (ret == -ENOMSG) {
+		return 0;
+	} else if (ret != 0) {
+		return ret;
+	}
+
+	struct le_audio_msg msg;
+
+	ret = zbus_chan_read(chan, &msg, K_NO_WAIT);
+
+	if (ret) {
+		return ret;
+	}
+	audio_streaming_event = msg.event;
+
+	switch (audio_streaming_event) {
+	case LE_AUDIO_EVT_STREAMING:
+		lv_label_set_text(play_pause_label, LV_SYMBOL_PAUSE);
+		break;
+
+	case LE_AUDIO_EVT_NOT_STREAMING:
+		lv_label_set_text(play_pause_label, LV_SYMBOL_PLAY);
+		break;
+	default:
+		break;
+	}
+	return 0;
+}
+
+static void volume_label_update(lv_obj_t *volume_label)
+{
+	double volume_level_int;
+	char volume_level_str[32];
+
+	volume_level_int = hw_codec_volume_get();
+	sprintf(volume_level_str, "%.0lf", floor((volume_level_int / 128) * 100));
```

Comment:
```suggestion
	sprintf(volume_level_str, "%.0lf%", floor((volume_level_int / 128) * 100));
```
I think this will work, then you can drop the `strcat` call

Reply:
I have tried this it does for some reason not work??